src/main/java/no/unit/nva/model/Publication.java

```diff
@@ -105,6 +105,26 @@ public void setDoiRequest(DoiRequest doiRequest) {
         this.doiRequest = doiRequest;
     }
 
+    /**
+     * Update a publication with the requested status change.
+     * @param requestedStatusChange Requested status change.
+     * @throws IllegalArgumentException Invalid status to change to.
+     * @throws IllegalStateException No DoiRequest exists.
+     * @see DoiRequestStatus
+     */
+    public void updateDoiRequestStatus(DoiRequestStatus requestedStatusChange) {
+        if (Objects.isNull(doiRequest)) {
+            throw new IllegalStateException(
+                ERROR_MESSAGE_UPDATEDOIREQUEST_MISSING_DOIREQUEST);
+        }
+        var ourDoiRequest = getDoiRequest();
```

Comment:
ourDoiRequest is the same as this.doiRequest. it is the same object as `this.doiRequest` i.e., `ourDoiRequest` and `this.doiRequest` point to exactly the same object. There is no point in having it. use `this.doiRequest`